src/crimson/os/seastore/onode_manager/staged-fltree/stages/stage_types.h

```diff
@@ -207,19 +208,20 @@ struct staged_position_t {
 
   index_t index;
   nxt_t nxt;
-};
-template <match_stage_t STAGE>
-std::ostream& operator<<(std::ostream& os, const staged_position_t<STAGE>& pos) {
-  if (pos.index == INDEX_END) {
-    os << "END";
-  } else if (pos.index == INDEX_LAST) {
-    os << "LAST";
-  } else {
-    os << pos.index;
-    assert(is_valid_index(pos.index));
+
+  std::string fmt_print() const {
+    std::string me_str;
+    if (index == INDEX_END) {
+      me_str = "END";
+    } else if (index == INDEX_LAST) {
+      me_str = "LAST";
+    } else {
+      assert(is_valid_index(index));
+      me_str = std::to_string(index);
+    }
+    return me_str + ", " + nxt.fmt_print();
```

Comment:
please don't. use `fmt::format_to()` instead.

Reply:
Will fix.

Reply:
@tchaikov : you meant 'fmt:;format()', right? I do not have a buffer here.

Reply:
we could provide a `string` instance, and feed it to `fmt::format_to()`.

Reply:
I'll try to measure the two alternatives. Thanks.

Reply:
This change was already tested - and now merged by Yuri. I will continue to test your idea as a possible improvement. Note that just using fmt::format into a local string is measured to be much slower than anything else, including slower than just concatenating strings. I will try to measure a version of the fmt_print that will have a string moved-in into it, than moved-out again.